engine/orchestration/src/com/cloud/vm/VirtualMachineManagerImpl.java

```diff
@@ -1472,6 +1473,12 @@ private void advanceStop(final VMInstanceVO vm, final boolean cleanUpEvenIfUnabl
                 _workDao.update(work.getId(), work);
             }
             return;
+        } else {
+            HostVO host = _hostDao.findById(hostId);
+            if (!cleanUpEvenIfUnableToStop && vm.getState() == State.Running && host.getResourceState() == ResourceState.PrepareForMaintenance) {
+                s_logger.debug("Host in PrepareForMaintenance state - Failed to stop VM with id: " + vm.getId());
```

Comment:
@pedro-martins The condition "host.getResourceState() == ResourceState.PrepareForMaintenance" only, in the if statement above, states that the Host is preparing for maintenance. Other conditions doesn't relate to it and so cannot go in "isHostPreparingForMaintenance" method.